frontend/src/pages/projects/screens/detail/storage/StorageTable.tsx

```diff
@@ -15,20 +19,66 @@ type StorageTableProps = {
 const StorageTable: React.FC<StorageTableProps> = ({ pvcs, refresh, onAddPVC }) => {
   const [deleteStorage, setDeleteStorage] = React.useState<PersistentVolumeClaimKind | undefined>();
   const [editPVC, setEditPVC] = React.useState<PersistentVolumeClaimKind | undefined>();
+  const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
+  const [alertStatuses, setAlertStatuses] = React.useState<Record<string, boolean>>({});
+
+  const isDeprecatedAlert = React.useMemo(
+    () => Object.values(alertStatuses).includes(true),
+    [alertStatuses],
+  );
```

Comment:
this whole alertStatuses is to just put the banner on the top. So instead of state, let us just have a useMemo that iterates over each pvc's storage class and returns true or false if to show the alert banner. Then you also dont need to have the setAlertStatuses anywhere as it reacts just to the pvc.

so something like 

```tsx
const showDeprecatedStorageClassAlert = React.useMemo(() => pvcs.some(pvc => // get storage class config and return true if deprecated ...
```